networkit/cpp/io/ThrillGraphBinaryWriter.cpp

```diff
@@ -103,11 +16,25 @@ void NetworKit::ThrillGraphBinaryWriter::write( const NetworKit::Graph &G, const
 					}
 			});
 
-			PutVarint(out_stream, neighbors.size());
-			
+			size_t deg = neighbors.size();
+
+			// Write variable length int for the degree
+			if(!deg) {
+				out_stream << uint8_t(0);
+			}
+
+			while(deg) {
+				auto u = deg & 0x7F;
+				deg >>= 7;
+				out_stream << uint8_t(u | (deg ? 0x80 : 0));
+			}
+
 			for (uint32_t v : neighbors) {
-				static_assert(std::is_same<uint32_t, decltype(v)>::value, "Node type is not uint32 anymore, adjust code!");
-				out_stream.write(reinterpret_cast<const char*>(&v), 4);
+				// write neighbor as little endian
+				for (size_t i = 0; i < sizeof(uint32_t); ++i) {
+					out_stream << uint8_t(v);
+					v >>= 7;
```

Comment:
This seems to be a typo. While we want to shift only 7 bits for varints, we're throwing away one bit of each byte here.

Reply:
Yes, this was a typo which is already fixed in 8ba00589be305a5a514ef68bfb10a3db047767ba.